functional-tests/tests/prover_el_deposit_withdraw.py

```diff
@@ -0,0 +1,178 @@
+            original_balance,
+            difference,
+            self.debug,
+        )
+
+        # Collect L2 and L1 blocks where the withdrawal has happened.
+        l2_withdraw_block_num = withdraw_tx_receipt["blockNumber"]
+        self.info(f"withdrawal block num on L2: {l2_withdraw_block_num}")
+
+        last_block_hash = btcrpc.proxy.getblockchaininfo()["bestblockhash"]
+        last_block = btcrpc.proxy.getblock(last_block_hash, 1)
+        # Check all blocks down from the latest.
+        # Those blocks will have only coinbase tx for all the empty blocks.
+        # Block with the withdrawal transfer will have at least two transactions.
+        while len(last_block["tx"]) <= 1:
+            last_block = btcrpc.proxy.getblock(last_block["previousblockhash"], 1)
+        l1_withdraw_block_height = last_block["height"]
+        self.info(f"withdrawal block height on L1: {l1_withdraw_block_height}")
+
+        # Proving
+        self.test_checkpoint(
+            l1_withdraw_block_height, l2_withdraw_block_num, prover_client_rpc, rethrpc
+        )
+
+    def test_checkpoint(self, l1_block, l2_block, prover_client_rpc, rethrpc):
+        self._chkpt_id += 1
+        l1 = (l1_block - 1, l1_block + 1)
+        l2 = (l2_block - 1, l2_block + 1)
+        # Wait some time so the future blocks in the batches are finalized.
+        # Given that L1 blocks are happening more frequent that L2, it's safe
+        # to assert only L2 latest block.
+        utils.wait_until(
+            lambda: int(rethrpc.eth_blockNumber(), base=16) > l2[1],
+            timeout=60,
+        )
+
+        task_ids = prover_client_rpc.dev_strata_proveCheckpointRaw(self._chkpt_id, l1, l2)
+
+        self.debug(f"got task ids: {task_ids}")
+        task_id = task_ids[0]
+        self.debug(f"using task id: {task_id}")
+        assert task_id is not None
+
+        wait_for_proof_with_time_out(prover_client_rpc, task_id, time_out=30)
+
+
+def confirm_btc_withdrawal(
+    withdraw_address,
+    btc_url,
+    btc_user,
+    btc_password,
+    original_balance,
+    expected_increase,
+    debug_fn=print,
+):
+    """
+    Wait for the BTC balance to reflect the withdrawal and confirm the final balance
+    equals `original_balance + expected_increase`.
+    """
+    # Wait for the new balance,
+    # this includes waiting for a new batch checkpoint,
+    # duty processing by the bridge clients and maturity of the withdrawal.
+    utils.wait_until(
+        lambda: get_balance(withdraw_address, btc_url, btc_user, btc_password) > original_balance,
+        timeout=60,
+    )
+
+    # Check final BTC balance
+    btc_balance = get_balance(withdraw_address, btc_url, btc_user, btc_password)
+    debug_fn(f"BTC final balance: {btc_balance}")
+    debug_fn(f"Expected final balance: {original_balance + expected_increase}")
+
+    assert (
+        btc_balance == original_balance + expected_increase
+    ), "BTC balance after withdrawal is not as expected"
```

Comment:
Don't we have something in `utils.py` already for this?
If not, this should be moved to `utils.py` anyways.

Reply:
done